### PR TITLE
Split Game Thread from Playing Field

### PR DESCRIPTION
**Pull request by Taeir**
- Moved the GameRunnable class out of the PlayingField, and created GameThread based on it.
- Changed Game Thread logic. Instead of recreating runnables, now only the thread is recreated when needed.

For reviewing this Pull Request: keep in mind that a lot of code was simply moved with minor changes applied.

<organisation

### CODE REVIEW

File: src/main/java/com/github/fishio/game/GameThread.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+				callPostTick("GameThread");
+				
+				end = System.currentTimeMillis();
+				
+				//Sleep if we have time to spare.
+				long dur = end - start;
+				if (dur < waitTime) {
+					try {
+						Thread.sleep(Math.round(waitTime - dur));
+					} catch (InterruptedException ex) { }
```
Comment (Diocruel) [inline src/main/java/com/github/fishio/game/GameThread.java]: _Auto-generated comment_

```
Must have at least one statement.
```

Again. Not sure if you need to change this. But it is on the same place as it was in the original playingfield class.

<layout

**Comment by Diocruel** [comment]
The rest looks good. It is nice for the gamethread to have it's own class now! Should really clean up PlayingField.

<layout,organisation

File: src/main/java/com/github/fishio/game/GameThread.java (hunk 1 above)
Comment (Taeir) [inline src/main/java/com/github/fishio/game/GameThread.java]: Checkstyle error didn't show up for me, good catch.
I fixed it in 284f755.

<layout

**Comment by LMiljak** [comment]
Pretty much most of the code remains the same, so I had no trouble reviewing it. I didn't notice any flaw in the code.
:+1: 

<review

**Pull request merged**